File: pyplayground/client/RobotThymio2.py
"""
Wrapper para el robot del tipo Thymio2

    from pyplayground.client import RobotFactory

    rob = RobotFactory.connectRobot( 'Thymio-01', host, port )
    rob.setSpeed( -1000, 1000 )
    sensores = rob.getSensors()
    rob.setLedIntensity( ( 3:0.5, 5:1.0 } )
"""

import socket

import pyplayground.utils.BasicSockJson as BasicSockJson
import pyplayground.client.RobotBase as RobotBase

class RobotThymio2( RobotBase.RobotBase ):
    def __init__( self, name:str, host:str, port:int, sock:socket.socket ):
        super().__init__( name, host, port, sock )

    def __str__( self ):
        return f'RobotThymio2 >> name:{self.name} - host={self.host} - port={self.port}'

    def setLedsIntensity( self, leds:dict ) -> None:
        """
        Cambia la intensidad de varios LEDs del robot

        Parameters
          leds: Un diccionario con los leds a afectar en su intensidad
                El indice corresponde al número del led
                El valor corresponde a su intensidad (float entre 0 y 1
        """
        pkg = { 'cmd':'setLedsIntensity', 'leds': leds }
        BasicSockJson.send( self.sock, pkg )
        resp = BasicSockJson.read( self.sock, self.buff )['answer']
        return None

File: java_client/src/main/java/rcr/jplayground/RobotThymio2.java
package rcr.jplayground;

import java.util.Map;
import java.util.ArrayList;

import org.json.*;

public class RobotThymio2 extends RobotBase {
    private float[] groundSensorValues = null;

    public RobotThymio2( String name, String host, int port ) throws Exception {
        super( name, host, port, "thymio2" );
    }

    public void getSensors()  throws Exception {
        Map<String, Object> map = retrieveSensors();
        groundSensorValues = getFloatArray( (ArrayList)map.get("groundSensorValues") );
    }

    public float[] getGroundSensorValues() {
        return groundSensorValues;
    }

    public void setLedsIntensity( float[] leds ) throws Exception {

        JSONObject json = new JSONObject();
        json.put( "cmd", "setLedsIntensity" );
        json.put( "leds", leds );
        out.print( json );
        out.print( '\n' );
        out.flush();

        String resp = in.readLine();
    }
}
